src/NServiceBus.Core/Serializers/XML/ReadOnlyMemoryStream.cs

```diff
@@ -1,16 +1,14 @@
-﻿// unset
-
-namespace NServiceBus
+﻿namespace NServiceBus
 {
     using System;
     using System.IO;
 
-    class ReadOnlyStream : Stream
+    sealed class ReadOnlyMemoryStream : MemoryStream
     {
-        ReadOnlyMemory<byte> memory;
-        long position;
+        readonly ReadOnlyMemory<byte> memory;
+        int position;
```

Comment:
Why this change? ReadOnlyMemory can't be longer than int.MaxValue bytes?

Reply:
Previously the implementation did cast the position to int on every Read call. This changed it to treat it int all the time on the field level but the outcome should be the same right?

Will double check though the runtime implementation again (I think I did it before but it doesn't hurt to be extra careful)

Reply:
@DavidBoike example https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/IO/MemoryStream.cs#L25-L27

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/IO/MemoryStream.cs#L315

It does a bit more validation but :shrug: